src/figuretype/trader.c

```diff
@@ -943,12 +966,33 @@ void figure_trade_ship_action(figure *f)
     f->image_id = image_group(GROUP_FIGURE_SHIP) + dir;
 }
 
+static double trade_policy_factor() {
+    double trade_factor = 1;
+
+    if(city_buildings_has_caravanserai() && building_monument_working(BUILDING_CARAVANSERAI)) {
+        building *b = building_get(city_buildings_get_caravanserai());
+
+        int policy = building_monument_module_type(BUILDING_CARAVANSERAI);
+
+        if (building_caravanserai_enough_foods(b) && policy == TRADE_POLICY_3) {
+            int pct_workers = calc_percentage(b->num_workers, model_get_building(b->type)->laborers);
+            if (pct_workers >= 100) { // full laborers
+                trade_factor = 1 + POLICY_3_BONUS_PERCENT/100; // caravan capacity increase by 50%
```

Comment:
Once again, no need to use a `double` and divide by `100` here. Just use `calc_adjust_with_percentage` where appropriate.